ball_detect/main.c

```diff
@@ -10,14 +12,17 @@
 #define BALL_SENSOR_3 2
 #define ULTRASONIC_SENSOR 4
 
-volatile int32_t pulse = 0;
+volatile int32_t pulse[4];
 volatile int32_t pulses[4]; // averages of pulses
-volatile int32_t counters[4]; // counters of lenghts of pulses
 volatile int32_t helper[4]; // lenghts of pulses
+volatile int32_t result[3]; //results of ball sensor visions :)
```

Comment:
Please rename this to `vision_results` or something appropriate and remove the comment.